### PR TITLE
Conventions on test scenarios files

### PR DESCRIPTION
**Pull request by antonioaversa**
Defines the conventions to follow when defining test scenarios files for C# rules.

Result of the circle around the ["Merge test cases depending on language version" initiative](https://trello.com/c/5uqnuqTM/76-merge-test-cases-depending-on-language-version).

### CODE REVIEW

File: docs/coding-style.md [hunk 1]
```diff
@@ -225,3 +225,13 @@ Do not use `ValueTuples` in production code. The usage in test projects is fine.
...
+
+## Test scenarios files
+
+For any C# rule `TheRule`, there should be at least two test scenarios files:
+* `TheRule.cs`, targeting both .NET Framework and .NET Core and using the default version of C#
+* `TheRule.Latest.cs`, targeting .NET Core only (via `# if NET`) and using the latest version of C#
+
+More test scenarios files can be created as needed, for code fixes, top-level statements etc.
+
+Test scenarios created in the past may still be split by language (e.g. `TheRule.CSharpX.cs`) and should be migrated following a Clean as You Code approach.
```
Comment (gregory-paidis-sonarsource) [inline docs/coding-style.md:237]: I would reword this, because I feel like it can be misunderstood as "you may still split the files".

I would rephrase it to:

In the past, test scenarios were split by language (e.g. `TestRule.CSharpX.cs`).
These files should be migrated, following a Clean as You Code approach.

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM, please take a look at the single comment I left.

File: docs/coding-style.md (hunk 1 above)
Comment (antonioaversa) [reply docs/coding-style.md:237]: Indeed! I haven't realized it could be read that way.
I have changed it, following your suggestion.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Test of GH "request changes". Ignore this.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
Undoing my test click. Ignore this.
